### DISCUSSION TITLE
CloudWatch DataSource: Support alias replacement for {{period}} in Expression field

### DISCUSSION THREAD

**Discussion by niklasR** [grafana/grafana#41811] [Data Sources]
_Similar to https://github.com/grafana/grafana/issues/20262_
**What would you like to be added**:
The CloudWatch datasource has internal variables that can be used for the "Alias" field, such as {{period}} or {{region}}.

We would like to be able to use those internal variables in the "Expression" field, too.

**Why is this needed**:
The DataSource provides computation of a relevant period for your chose time window, but this is only usable when querying metrics directly, not in a Search expression. In a Search expression, the period always needs to be hard-coded or be based on a Dashboard variable, which is limiting.

**Comment by rehevkor5** [comment]
Related: https://github.com/grafana/grafana/discussions/44489

**Comment by sarahzinger** [comment]
@niklasR Apologies, I know it's been a long time since you started this discussion but I believe that this may have been fixed, but let me know if we've missed something. 

I see in our [documentation](https://grafana.com/docs/grafana/latest/datasources/aws-cloudwatch/query-editor/#period-macro) support for `$__period_auto` in Search expressions. 

We are in the process of moving from [discussions to issues](https://github.com/grafana/grafana/discussions/73424) so I'm going to close this discussion for now but if there are macros you'd like to see that we do not have (or have limited support for), feel free to open an issue and we'll make sure to consider if for future work. 

Thanks for the suggestion!
